### PR TITLE
h803x: Initial implementation

### PR DESCRIPTION
**Pull request by jontrulson**
This module implements support for the Veris H8035 and H8036 Energy
Meters.

The H8036 is similar to the H8035, but provides much more data.

The Enercept H8035/H8036 is an innovative three-phase networked
(Modbus RTU) power transducer that combines electronics and high
accuracy industrial grade CTs in a single package. The need for
external electrical enclosures is eliminated, greatly reducing
installation time and cost. Color-coordination between voltage leads
and CTs makes phase matching easy. Additionally, these transducers
automatically detect and compensate for phase reversal, eliminating
the concern of CT load orientation. Up to 63 Transducers can be
daisy-chained on a single RS-485 network.

This module was developed using libmodbus 3.1.2, and the H8035.  The
H8036 has not been tested.  libmodbus 3.1.2 must be present for this
module to build.

Signed-off-by: Jon Trulson <jtrulson@ics.com>

### CODE REVIEW

File: examples/c++/CMakeLists.txt [hunk 1]
```diff
@@ -240,6 +240,7 @@ add_example (rhusb)
 add_example (apds9930)
 add_example (kxcjk1013)
 add_example (ssd1351)
+add_example (h803x)
```
Comment (Propanu) [inline examples/c++/CMakeLists.txt]: This needs to go into the Modbus block.

Comment (jontrulson) [inline examples/c++/CMakeLists.txt]: Yes it does.  Will fix and build test.

Comment (jontrulson) [inline examples/c++/CMakeLists.txt]: tested, and repushed.

**Comment by Propanu** [comment]
Thank you!

**Pull request closed without merging**